src/main/java/com/uber/cadence/client/WorkflowStub.java

```diff
@@ -156,27 +155,38 @@
    */
   <R> R query(String queryType, Class<R> resultClass, Object... args);
 
+  /**
+   * Deprecated: please use {@link #queryWithOptions(String, QueryOptions, Type, Class, Object...)}
+   * to avoid variable argument ambiguity with Object... args
+   */
   <R> R query(String queryType, Class<R> resultClass, Type resultType, Object... args);
 
+  /**
+   * Deprecated: please use {@link #queryWithOptions(String, QueryOptions, Type, Class, Object...)}
+   * to avoid variable argument ambiguity with Object... args
+   */
   <R> R query(
       String queryType,
       Class<R> resultClass,
       QueryRejectCondition queryRejectCondition,
       Object... args);
 
+  /**
+   * Deprecated: please use {@link #queryWithOptions(String, QueryOptions, Type, Class, Object...)}
+   * to avoid variable argument ambiguity with Object... args
+   */
   <R> R query(
       String queryType,
       Class<R> resultClass,
       Type resultType,
       QueryRejectCondition queryRejectCondition,
       Object... args);
 
-  <R> R query(
+  <R> R queryWithOptions(
       String queryType,
-      Class<R> resultClass,
+      QueryOptions options,
       Type resultType,
-      QueryRejectCondition queryRejectCondition,
-      QueryConsistencyLevel queryConsistencyLevel,
+      Class<R> resultClass,
```

Comment:
to reviewer: I put `Class<R> resultClass` right before `Object... args` so that there is no ambiguity